### ISSUE TITLE
CA1004: allow to disable the rule for each field

### ISSUE THREAD

**Issue by curia-damiano** [dotnet/roslyn-analyzers#4385]
### Analyzer

**Diagnostic ID**: [CA1716](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1716)

### Describe the improvement
The rule CA1004 reports classes that contain fields are that are IDisposable and are not properly disposed.
In case of suppression of the rule, it gets suppressed for the full class.
It would be nice if:
- it could keep the current behavior, when applied to a whole class. I.e. suppressing at class level will ignore any field in the class itself;
- but also introduce the possibility to suppress it at field level, so I can ignore one field, but still keep the rule valid for the class, so if later another developer will introduce a new IDisposable field, it will be correctly picked up by the rule.

The improvement would be the possibility to suppress the rule at field level.

**Comment by mavasani** [comment]
This can potentially be implemented by adding support for the existing editorconfig option https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#excluded-symbol-names for this rule. Tagging @Evangelink in case you are interested in picking this up.

**Comment by Evangelink** [comment]
@curia-damiano are you referring to [CA2213](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2213)? I could not find `CA1004`. If that's the case, what is preventing to use the following syntax:

```csharp
public sealed class C : IDisposable
{
	public void Dispose()
	{
	}
}

public sealed class B : IDisposable
{
#pragma warning disable CA2213 // Disposable fields should be disposed
	private readonly C c = new C();
#pragma warning restore CA2213 // Disposable fields should be disposed

	public void Dispose()
	{
	}
}
```

**Comment by curia-damiano** [comment]
Hi,
Sorry, the warning is about rule CA1001 - https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001.

When I compile, I get a message of this type:
  "Type 'MyType' owns disposable field(s) 'Field1, Field2' but is not disposable"

The point is that if I try to suppress, this rule, it is suppressed at class level:

#pragma warning disable CA1001 // Types that own disposable fields should be disposable
    public class MyType
#pragma warning restore CA1001 // Types that own disposable fields should be disposable

If I move the suppression and re-enabling pragmas to the specific field:

#pragma warning disable CA1001 // Types that own disposable fields should be disposable
        public XXX Field1;
#pragma warning restore CA1001 // Types that own disposable fields should be disposable
        public XXX Field2;

I still get the warning on both fields.

I would not like to suppress the message at class level, because in fact I would not be able to detect an issue with a potential second field. As said, I would like to suppress the rule for one field only.

**Comment by Evangelink** [comment]
Ok makes sense now! I will try to handle this as suggested by @mavasani

**Comment by curia-damiano** [comment]
Thank you @Evangelink
